src/components/Node/Node.tsx

```diff
@@ -42,11 +42,15 @@ const Node: FC<NodeProps & { ref: Ref<HTMLDivElement> }> = memo(
         left: xCoordinate
       };
 
+      const isCurrentNode = pathwayState.key === currentNode?.key;
```

Comment:
On line 26 currentNode is defined as 
`currentNode: state`
So it should never be null. Therefore you won't need the null check here
```suggestion
      const isCurrentNode = pathwayState.key === currentNode.key;
```